07-generic_bookworms/final/collectors/load_internal_test.go

```diff
@@ -0,0 +1,57 @@
+package collectors
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func noError(t *testing.T, err error) {
+	t.Helper()
+	assert.NoError(t, err)
+}
+
+func TestLoad(t *testing.T) {
+	tests := map[string]struct {
+		collsFile string
+		want      Collectors[item]
+		checkErr  func(*testing.T, error)
+	}{
+		"file exists": {
```

Comment:
If you need checks that have both succeeding and failing cases, then they probably should be separate tests altogether.